src/t_hash.c

```diff
@@ -1398,7 +1401,9 @@ int hashTypeNext(hashTypeIterator *hi, int skipExpiredFields) {
         while ((hi->de = dictNext(hi->di)) != NULL) {
             hi->expire_time = hfieldGetExpireTime(dictGetKey(hi->de));
             /* this condition still valid if expire_time equals EB_EXPIRE_TIME_INVALID */
-            if (skipExpiredFields && ((mstime_t)hi->expire_time < commandTimeSnapshot()))
+            if (!server.allow_access_expired &&
```

Comment:
@moticless please help confirm whether there will be any side effects here. I'm worried that skipExpiredFields might conflict with allow_access_expired.

Reply:
please don't use force push. it's hard to review.

Reply:
@moticless please take a look this comment.

Reply:
This commit only fixes genericHgetallCommand(), but many other places call hashTypeLength() and incorrectly subtract expired fields when server.allow_access_expired is enabled.

hashTypeLength() should check server.allow_access_expired internally rather than relying on callers to pass the correct subtractExpiredFields parameter. We need a test to verify all hash commands behave consistently with this flag.

Reply:
Yes, they're functionally equivalent, but the current approach is error-prone. Every caller must remember to check server.allow_access_expired and pass the correct parameter. This is exactly what's causing bugs.

If hashTypeLength() checks server.allow_access_expired internally, there's no ambiguity - it always behaves consistently. This follows the same pattern as keyIsExpired(), hashTypeIsExpired(), and hfieldIsExpired() which already check the flag internally.